jep/0000/README.adoc

```diff
@@ -0,0 +1,345 @@
+For this example plugin, Karl has set the value to `3`, because he's usuing semantic versioning, e.g., 
+3.1.4.
+
+==== Maven POM file changes
+
+Karl's plugin is already making use of Incrementals, and the relevant lines of his `pom.xml` file 
+look like:
+
+```
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.jenkins-ci.plugins</groupId>
+        <artifactId>plugin</artifactId>
+        <version>4.56</version>
+        <relativePath />
+    </parent>
+    <artifactId>karl</artifactId>
+    <version>${revision}${changelist}</version>
+    <packaging>hpi</packaging>
+    <name>Karl Plugin</name>
+    <url>
+        <!--Something like https://wiki.jenkins-ci.org/display/JENKINS/Karl+Plugin-->
+    </url>
+    <description>A useful description of the Karl plugin.</description>
+    <licenses>
+        <license>
+            <name>MIT</name>
+            <url>http://opensource.org/licenses/MIT</url>
+        </license>
+    </licenses>
+
+    <properties>
+        <revision>3.1.4</revision>
+        <changelist>-SNAPSHOT</changelist>
+        <java.level>8</java.level>
+        <jenkins.version>2.138.4</jenkins.version>
+    </properties>
+
+```
+
+No additional changes need to be made to `pom.xml` by Karl, he's good to go.
+
+==== Autorelease.yaml verification in `buildPlugin`
+
+The commonly used library link:https://github.com/jenkins-infra/pipeline-library/blob/master/vars/buildPlugin.groovy[`buildPlugin`] will need to be modified to check for the presence of, and validate, `Autorelease.yaml`. This 
+markerfile will be the mechanism that tells `${trusted-ci-server}` that this plugin should be automatically 
+released. 
+
+Validation must include, but will not be limited to, the following:
+
+1. `pluginName` field must match the repository name
+
+2. `versionNumberFields` must match the `revision` property in `pom.xml`. E.g., if `versionNumberFields` is 
+set to `3` (three fields), and the `revision` in `pom.xml` is set to `3.1.4.5` (four fields), validation 
+will fail.
+
+3. Validation that no extra fields are present in the file. Comments are allowed, but any unexpected 
+lines will cause validation to fail.
+
+If validation of `Autorelease.yaml` fails for any reason, the build is not performed nothing gets deployed, 
+and GitHub is notified of the failure.
+
+==== Deployment of a successful release
+
+When Karl merges a commit into the `master` branch, that merge commit is commit number 150, and has the 
+SHA `1a2b3c4`. The following takes place:
+
+* A webhook is sent to `${trusted-ci-server}`, and `Autorelease.yaml` is validated.
+* Once validation passes, a build is performed. If the build passes all its tests, a release 
```

Comment:
This build is happening on trusted.ci.jenkins.io or whatever, in addition to the build happening on ci.jenkins.io? Or is ci.jenkins.io the “trusted CI server”?

Reply:
Because the build is getting triggered by a webhook set up specifically for this purpose, the CD builds could be built only on _trusted.ci.jenkins.io_, bypassing the building it twice problem.

Reply:
ci.jenkins.io is not involved here at all (except for a CI build of the master branch).

trusted.ci.j.io would be possible, but that instance is a legacy mess.

Would rather go with a shiny new autorelease.ci.j.io on the Jenkins infra VPN.

Reply:
> except for a CI build of the master branch

That is what I just said. You push to `master` and two Jenkins services initiate builds. Or am I missing something?

Reply:
OK, then we agree here.

Reply:
```suggestion
* Once validation passes, a build is performed. If the build passes all its tests, a release 

WARNING: Some clarification needed based on https://github.com/jenkinsci/jep/pull/244/files#r293536274 .

```